src/EnergyPlus/WeatherManager.cc

```diff
@@ -6921,6 +6921,13 @@ namespace Weather {
             } break;
             } // switch
 
+            if (!ipsc->lNumericFieldBlanks(3)) {
+                state.dataWeather->WaterMainsTempsMultiplier = NumArray(3);
+            }
+            if (!ipsc->lNumericFieldBlanks(4)) {
+                state.dataWeather->WaterMainsTempsOffset = NumArray(4);
+            }
```

Comment:
Pointing out that when using getObjectItem to process the object, blank fields that have a default will return the default. You can easily test this in the debugger. If these fields are blank then NumArray(3) = 1 and NumArray(4) = 0. So you could just read the inputs directly without checking if they are blank.

Reply:
Thanks, @rraustad. I had that originally, but added these protections (that I saw used elsewhere in other classes) after getting a crash. But it turns out the crash was because of [this](https://github.com/NREL/EnergyPlus/pull/11037/commits/09f8a76bfc6b2221c532067125f8747cbf0bb76c), so I will revert these protections.

No idea why this class has arrays that need to be dimensioned before `getObjectItem` is called, most other classes don't have that. But whatever...